translator/translate/otel/pipeline/jmx/translator.go

```diff
@@ -104,6 +105,9 @@ func (t *translator) Translate(conf *confmap.Conf) (*common.ComponentTranslators
 	if conf.IsSet(common.ConfigKey(common.MetricsKey, common.AppendDimensionsKey)) {
 		translators.Processors.Set(ec2taggerprocessor.NewTranslator())
 	}
+	if context.CurrentContext().RunInContainer() {
```

Comment:
Is there a reason we didn't use the above `if context.CurrentContext().RunInContainer()`?